Added latitude and longitude to location markers

kevinpark1217 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

Displays location in latitude and longitude to all wild Pokemon, gym,
and PokeStops in the marker bubble.
Also, added a gym level indicator.

## Description
<!--- Describe your changes in detail -->

It allows easy view of the location coordinates for each location marker. It makes it easy to find and type in the coordinates to any other third party applications.
Ex: GPS Spoofer

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
It is no longer necessary to press "Inspect Element" to find the location coordinates.
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

I have tested this on my Ubuntu Server and ran as expected without any problems.
This only adds couple lines to the javascript file; therefore, it was unnecessary to test it more than once. 

## Screenshots:

![Pokemon Screenshot](https://i.gyazo.com/139e1e9d924cf2ad520375dec15d4761.png)
![Gym Level Indicator](https://i.gyazo.com/8c226b18d11c989945f4cd7eade8ee51.png)

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ x] I have updated the documentation accordingly.

callmebackdraft commented:
This is quite a nice addition, however am i the only one who thinks precaution should be taken not to flood the infowindows with alot of info, seeing as they are getting quite big.

kevinpark1217 commented:
@callmebackdraft That is what I was thinking about the gym label because there are a lot of things including the logo. However for wild pokemons, four lines like the screenshot seems reasonable.

I am also considering adding a gym level right next to the gym prestige, so it doesn't take any additional lines.

Idhanta commented:
+1

MonkeyLeeT commented:
Tested and working. But ideally it should be off by default (page performance was impacted), can you make some improvement?

kevinpark1217 commented:
@MonkeyLeeT Could you please tell me how much of the page performance was impacted? And how did you test it? I would like to test the page performance myself and find out where the problem is.

I don't know where the performance would be greatly impacted other than the javascript toFixed() method.

MonkeyLeeT commented:
Oh it was not greatly impacted, it's just http://www.webpagetest.org/ to test some basic metrics. Though small, at this point any changes that might add load to page rendering should be made optional.

kevinpark1217 commented:
@MonkeyLeeT 
I believe that your test is flawed. The test results varied quite heavily depending on what point of time the browser requested for the website. For example, the webserver would take a little longer to respond during a scan rather than a sleep. Therefore, the conclusion is that my changes are too small to be negligible in page rendering process.

I have tested one with my changes first, then the one without my changes. The results are as it follows.
[With Changes](http://www.webpagetest.org/result/160724_8G_72V/)
[Without Changes](http://www.webpagetest.org/result/160724_8J_76Y/)

I have checked multiple times to make sure that I have not mixed up my results.
Hopefully, this resolves the issue.

MonkeyLeeT commented:
@kevinpark1217 I agree with you. +1 from me.

kevinpark1217 commented:
@MonkeyLeeT So could this be merged with the develop branch?

Chlodochar commented:
Please match the indentation of the surrounding lines. You're using tabs, the surrounding lines have spaces.

MonkeyLeeT commented:
@kevinpark1217 please makes changes as suggested above and we'll merge.

kevinpark1217 commented:
@Chlodochar @MonkeyLeeT I have fixed the indentation problem. So, it is ready to merge.

Meanwhile, I have added a gym level indicator, which I have already tested. And it will look like following.
![Gym Level Indicator](https://i.gyazo.com/8c226b18d11c989945f4cd7eade8ee51.png)

MonkeyLeeT commented on static/map.js:
```diff
@@ -208,19 +212,30 @@ function gymLabel(team_name, team_id, gym_points, latitude, longitude) {
             </div>
             </center></div>`;
     } else {
+    	var gym_prestige = [2000, 4000, 8000, 12000, 16000, 20000, 30000, 40000$
```
40000$?

MonkeyLeeT commented:
Sorry to bother you with one more comments. This should be last.

kevinpark1217 commented:
@MonkeyLeeT Okay. I think I caught them all now. No pun intended.

MonkeyLeeT commented:
Merged!
